lib/msal-common/src/url/UrlString.ts

```diff
@@ -40,10 +40,16 @@ export class UrlString {
     static canonicalizeUri(url: string): string {
         if (url) {
             url = url.toLowerCase();
-        }
 
-        if (url && !StringUtils.endsWith(url, "/")) {
-            url += "/";
+            if (StringUtils.endsWith(url, "?")) {
+                url = url.slice(0, -1);
+            } else if (StringUtils.endsWith(url, "?/")) {
+                url = url.slice(0, -2);
+            }
```

Comment:
I'm curious about the intent here?  The example that I gave in #2705 only had a `?/` because this function (canonicalilzeUrl) added a `/` after the `?`.  Under what other circumstances would a `?/` need to be removed?